apollo-ios/Sources/Apollo/GraphQLResult.swift

```diff
@@ -47,3 +47,28 @@ extension GraphQLResult: Equatable where Data: Equatable {
 }
 
 extension GraphQLResult: Hashable where Data: Hashable {}
+
+extension GraphQLResult {
+  public func asDictionary() -> [String: Any] {
```

Comment:
I wonder if `asDictionary` is clear enough. Everyone that's asked for this has done it in reference to wanting "JSON" output, would 'asJSON' be more intuitive and result in less follow-up questions?

Reply:
Hmmm maybe even more verbose and say `asJSONDictionary`? I would think JSON would be clear enough but I guess you never know.

Reply:
I agree. It should have `JSON` in the name. I'm wondering if we should actually just make `GraphQLResult` and `GraphQLError` conform to `JSONEncodable` and then make this a `var _jsonValue`. But that would return a `[String: AnyHashable]`. @jimisaacs would that be alright for you, or do you prefer it to return a `[String: Any]`?

Reply:
Just as long as it's simple to serialize to Foundation.Data, and marshal back into generated types, I'm good. My original public interface returned Foundation.Data.

Reply:
I personally think just leaving it as [String: Any] and adding JSON to the name is probably fine. Enforcement of AnyHashable feels unnecessary since encoders all have to check every value and throw anyway.